api/v1alpha1/etcdcluster_types.go

```diff
@@ -200,6 +203,41 @@ type StorageSpec struct {
 	VolumeClaimTemplate EmbeddedPersistentVolumeClaim `json:"volumeClaimTemplate,omitempty"`
 }
 
+// SecuritySpec defines security settings for etcd.
+// +k8s:openapi-gen=true
+type SecuritySpec struct {
+	// +optional
+	Peer *PeerSpec `json:"peer,omitempty"`
+	// +optional
+	ClientServer *ClientServerSpec `json:"clientServer,omitempty"`
+}
+
+type PeerSpec struct {
+	// +optional
+	Ca SecretSpec `json:"ca,omitempty"`
+	// +optional
+	Cert SecretSpec `json:"cert,omitempty"`
+}
+
+type ClientServerSpec struct {
+	// +optional
+	Ca SecretSpec `json:"ca,omitempty"`
```

Comment:
This field needs a comment, documenting its purpose, e.g.
```go
  // Ca is the trusted CA used to sign client certificates...
  // +optional
  Ca SecretSpec `json:"ca,omitempty"`
  ```
  and in line with etcd's terminology `TrustedCA` might be a better name for it.

Reply:
We look here not only into etcd configuration, but also in the operator-managed certificates generation. That is why it is not only TRUSTED CA certificate, but also CA.KEY that signs client certificates.

For me TRUSTED means certificate without a key.